aws-controllers-k8s/sagemaker-controller
# Feature: Serverless Endpoint

Issue #, if available:

Description of changes:

This pull request contains code/unit tests for the Serverless Endpoint.
Changes made:
`generator.yaml` - Uncommenting shape and adding hook.
`pkg/resource/endpoint_config/custom_delta.go` - Code that handles server side default. Late initialization isnt possible because this field is an array.
`testdata/` - Unit tests

The Server returns a value for VolumeSizeInGB despite it being a parameter the user cannot even specify while using a serverless endpoint. The same problem is not there for an instance based endpoint config.

Manual testing (Will delete this before merge):

Endpoint Config:

[Serverless] Async Conifg - Validation Exception
[Serverless] Multi Variant - Validation Exception
[Serverless] Data Capture - Validation Exception
[Serverless] Including VolumeSizeInGB in spec: Validation exception
[Serverless] No VolumeSizeInGB: Normal reconciliation, no errors after restart.
[Normal] No VolumeSizeInGB - Normal reconciliation, no errors after restart.
[Normal] VolumeSizeInGB - Normal reconciliation, no errors after restart.

Endpoint:
Launching Serverless: Works, user sees information in PendingDeploymentSummary
Updating Instance→ Serverless: Validation Exception
Updating Serverless → Instance: Production variant must have a different name, works.


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
